harvard-lts/drs-translation-service
# Integrates Batch Builder

**Integrates and runs batch builder**
* * *

**GitHub Issue**: https://github.com/harvard-lts/HDC/issues/216

# What does this Pull Request do?
Finishes directory formatting
Integrates and runs Batch Builder and cleans up the files where necessary.
Places the LOADING file once batches complete.

# How should this be tested?
Please note the new environment variables in this release (see the env-template.txt for comparison)

1. Clone the repo
2. Checkout the branch HDC_216
3. Copy the Batch Builder zip file (see email) into the `batch_builder_client` directory and unzip.
4. Copy the env-template.txt to .env
5. Get the credentials from LPE Shared-HDC/Dev/QA ims ActiveMQ (transfer and process) for ims-process-dev
6. Add the credentials from step 4 to the .env for the PROCESS queueing and DRS 
7. Set .env variables:
```
DRS_QUEUE_CONSUME_NAME=/queue/mock-drs-ingest-complete-local
DRS_QUEUE_PUBLISH_NAME=/queue/mock-drs-ingest-trigger-local
PROCESS_QUEUE_CONSUME_NAME=/queue/dims-data-ready-mock-local
PROCESS_QUEUE_PUBLISH_NAME=/queue/drs-ingest-status-local
```
8. Start up the docker container: `docker-compose -f docker-compose-local.yml up -d --build --force-recreate`
9. Exec into the container: `docker exec -it translation-service bash`
10. Run the command `pytest`

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? yes
- integration tests? no

# Interested parties
@awoods @jessjass
